api/app/DoctrineMigrations/Version237.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Application\Migrations;
+
+use Doctrine\DBAL\Schema\Schema;
+use Doctrine\Migrations\AbstractMigration;
+
+/**
+ * Auto-generated Migration: Please modify to your needs!
+ */
+final class Version237 extends AbstractMigration
+{
+    public function getDescription() : string
+    {
+        return 'Create court_order rows for all existing lay clients';
+    }
+
+    public function up(Schema $schema) : void
+    {
+        // this up() migration is auto-generated, please modify it to your needs
+        $this->abortIf($this->connection->getDatabasePlatform()->getName() !== 'postgresql', 'Migration can only be executed safely on \'postgresql\'.');
+
+        $this->addSql("
+insert into court_order (client_id, type, supervision_level, order_date, case_number)
+select 
+  distinct(c.id),
+  case
+    -- if client does not have a report (only has an NDR) then type at this point is NULL
+    when r.id is null then null
+    -- otherwise assign HW for any '4' based reports, and PFA for all others
+    when r.type = '102_4' 
+      or r.type = '103_4' 
+      or r.type = '104' 
+      then 'HW'
+    else 'PFA'
+  end,
+  case
+ 	when r.type = '102' or r.type = '102_4' then 'GENERAL'
+ 	when r.type = '103' or r.type = '103_4' then 'MINIMAL'
+ 	-- null for HW only (104) 
+ 	else null
+  end,
+  case
+    -- can be nullable as some clients may no longer be in casrec table but we still want a court_order created
+    when ca.order_date is not null then ca.order_date
+    else null
+  end,  
+  c.case_number
+from client c
+-- left join as some clients may not have a report but an odr instead
+-- a client likely has > 1 report but we only want 1 report in order to read the r.type - using MAX(due_date) as selection criteria
+left join report r on r.client_id = c.id and r.due_date = (select MAX(due_date) from report where client_id = c.id)
```

Comment:
This is cool. Another way of doing this without the sub join would be using ROW_NUMBER() and partition by client_id, order by due_date. Just a bit of sql fun to look in to for another time. I'm not sure which is quicker to be honest.

Reply:
Oh nice